src/pages/collection/series/TmdbLinking.tsx

```diff
@@ -195,9 +195,17 @@ const TmdbLinking = () => {
   const createEpisodeLinks = useEventCallback(async () => {
     setCreateInProgress(true);
     try {
-      if (isNewLink && Object.keys(linkOverrides).length === 0) {
+      // If we're not giving the server any clues about which series to link
+      // then we need to first create the auto links before sending the
+      // mappings.
+      if (
+        isNewLink
+        && (Object.keys(linkOverrides).length === 0 || every(linkOverrides, links => every(links, link => link === 0)))
+      ) {
         await createAutoLinks({ tmdbShowID: tmdbId });
-      } else if (Object.keys(linkOverrides).length > 0) {
+      }
+
+      if (Object.keys(linkOverrides).length > 0) {
```

Comment:
Can this not be "else"?
When both the conditions are false, the flow shouldn't even end up here. The save link button should be disabled

Reply:
we only need to send the create links commands if there are no overrides, but we need to send both commands if we have overrides and all of them are set to id 0, since otherwise the server won't be able to know which tmdb show it should link to. if we have any overrides that are not id 0 then we only need to send the edit mapping command. also this is only for the initial link, as subsequently edits to existing links once the link has been established aren't effected even if we only link to id 0.

Reply:
Makes sense 😔